libbeat/common/config.go

```diff
@@ -190,6 +202,39 @@ func (c *Config) SetChild(name string, idx int, value *Config) error {
 	return c.access().SetChild(name, idx, value.access(), configOpts...)
 }
 
+func (c *Config) IsDict() bool {
+	return c.access().IsDict()
+}
+
+func (c *Config) IsArray() bool {
+	return c.access().IsArray()
+}
+
+func (c *Config) DebugString() string {
+	var bufs []string
+	if c.IsDict() {
+		var content map[string]interface{}
+		if err := c.Unpack(&content); err != nil {
+			return fmt.Sprintf("<config error> %v", err)
+		}
+		j, _ := json.MarshalIndent(content, "", "  ")
```

Comment:
tl;dr debug output is already a subset of YAML :P 

I opted for json in debug output, so it's more easy to 'see' the config as is loaded. e.g. in json it gets more apparent if users got the alignment wrong. Plus one can simply copy the json (even if debug log is not properly formatted on disucss) and use the debug config in place of the yaml (as JSON is a subset of YAML).